packages/react-native/ReactCommon/react/performance/timeline/PerformanceEntryLinearBuffer.cpp

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+*/
+
+#include "PerformanceEntryLinearBuffer.h"
+#include <string>
+
+namespace facebook::react {
+
+void PerformanceEntryLinearBuffer::add(const PerformanceEntry& entry) {
+  entries_.push_back(entry);
+}
+
+void PerformanceEntryLinearBuffer::consume(std::vector<PerformanceEntry>& target) {
+  target.reserve(entries_.size());
```

Comment:
This is just reserving the size but it's actually not copying the data into the target.